libvips/conversion/composite.cpp

```diff
@@ -194,8 +193,47 @@ typedef struct {
 	 */
 	VipsPel **p;
 
+#ifdef HAVE_VECTOR_ARITH
+	/* max_band as a vector, for the RGBA case.
+	 */
+	v4f max_band_vec;
+#endif /*HAVE_VECTOR_ARITH*/
```

Comment:
How about putting this struct member first?

At the moment, we could perhaps change the size of one the members before this and break alignment. I think it only works at the moment because of the exact number of pointers we have.

A comment about the need to have the vector as the first element would probably be useful.

Reply:
Did you mean to put this member first within the `VipsCompositeSequence` or move it back to `VipsCompositeBase` and put it there as first? I tried the latter version (see commit https://github.com/kleisauke/libvips/commit/dcee3d11fe701c854b8e7ce1964f6ee0951f0b70, which is the smallest change), but that didn't seem to work.

Do you think this could work without those aligned malloc implementations?

Reply:
I meant first in `VipsCompositeSequence`. 

My thinking was that the aligned malloc guarantees that the start of the struct will be on a 16-byte boundary, but `max_band_vec` will be aligned on something like (pointer-to-base + 3 * sizeof(pointer) + sizeof(int) + 2 * sizeof(pointer)). If we put the struct members with alignment requirements at the start of the struct, we can guarantee they will stay aligned even if the struct changes.

I'm not sure I understand how these structs are being laid out :( You'd think this would be:

```
base: composite
base + 4: input_regions
base + 8: composite_regions
base + 12: n
base + 16: enabled
base + 20: p
base + 24: max_band_vec
```

Which would obviously break alignment. I guess some padding somewhere must be making max_band_vec start at byte 32.

Reply:
Ah yes, I'm also not entirely sure how these structs are laid out, so moving the `max_band_vec` member to the first position in the struct makes sense. Done with commit 305714c9 (tested on Windows 32-bit without any problems).

fwiw, I think the above mentioned commit also reproduced the crash on Linux x64, see:
https://travis-ci.org/github/kleisauke/libvips/jobs/764105312#L2420